lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1453,13 +1453,16 @@ def _parse_publish_ports(self):
                 port_binds = [(default_ip, port) for port in parse_port_range(parts[0], self.client)]
             elif p_len == 3:
                 # We only allow IPv4 and IPv6 addresses for the bind address
-                if not re.match(r'^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$', parts[0]) and not re.match(r'^\[[0-9a-fA-F:]+\]$', parts[0]):
+                ipaddr = parts[0]
+                if not re.match(r'^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$', parts[0]) and not re.match(r'^\[[0-9a-fA-F:]+\]$', ipaddr):
```

Comment:
Note, not a blocker for this backport but a few things to keep in mind:
* ````re.compile()```'ing these regexes outside of the loop would be a nice optimization
* there may be something that's useful in ```ansible.module_utils.compat.ipaddress``` that would let us not maintain our own regexes for some of the things done here.